### DISCUSSION TITLE
bug fix. Remove the indentation at the beginning

### DISCUSSION THREAD

**Discussion by TheOnlyFastCoder2** [PrismJS/prism#3370] [Q&A]
Step 1)
Open the file prism.js and looking for ( ctrl + F  )  innerHTML   and add method trim() to the  variable
Step 2)
Open the css file with styles for prism and looking for pre[class*=language-] and add option "white-space: nowrap;"
![image](https://user-images.githubusercontent.com/60759188/156928169-28deb38b-b0f5-4380-a7c1-63fc7d832d0f.png)
![image](https://user-images.githubusercontent.com/60759188/156927932-4b8830c0-9096-4096-b801-80180a212015.png)

Example bug:
![image](https://user-images.githubusercontent.com/60759188/156928401-1abf07d4-9f6c-4ed5-a894-ac4a5d228fe3.png)

**Comment by RunDevelopment** [comment]
Prism (without any plugins) guarantees that the `textContent` of the highlighted element does not change. Prism does not add or remove indentation. So all the spaces (or tabs) you identified as a bug were there before Prism highlighted the code.

If you want Prism to remove/normalize the indentation of the input code, checkout the [Normalize Whitespace plugin](https://prismjs.com/plugins/normalize-whitespace/) and its `remove-initial-line-feed` option.

---

I also want to point out that your example is incomplete. We need to be able to see the HTML code of the code block with the bug. We cannot reproduce and fix any issues otherwise.
